inst/include/dust/dust.hpp

```diff
@@ -286,11 +299,31 @@ class Dust {
     _rng.long_jump();
   }
 
+  void set_data(std::map<size_t, data_t> data) {
+    _data = data;
+  }
+
+  std::vector<double> compare_data() {
+    std::vector<double> res;
+    auto d = _data.find(step());
+    if (d != _data.end()) {
```

Comment:
What about the else case here? Does the interface turn this uninitialised vector into a null for us?

Reply:
We get a vector of length 0, which is tested against in the interface and returns `NULL` back to R - that ends up being the same behaviour as the mcstate R compare version

Reply:
However, that's not at all obvious so I will add a comment :)